content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.set('listId', FlowRouter.getParam('_id'));
+    this.subscribe('list/todos', this.state.get('listId'), this.state.get('requestedTodos'));
+  });
+});
+```
+
+We'd increment that `requestedTodos` variable when the user clicks "load more" (or perhaps just when they scroll to the bottom of the page).
+
+Once piece of information that's very useful to know when paginating data is the *total number of items* that you could see. The `[tmeasday:publish-counts](https://atmospherejs.com/tmeasday/publish-counts)` can be useful to publish this. We could add a `/list/todoCount` publication like so
+
+```js
+Meteor.publish('list/todoCount', function(listId) {
+  check(listId, String);
+  
+  Counts.publish(this, `list/todoCount${listId}`, Todos.find({listId}));
+});
+```
+
+Then on the client, after subscribing to that publication, we can access the count with `Counts.get(``list/todoCount${listId}`)`.
+
+## Other (non-published) data sets: Stores
+
+In Meteor, persistent or shared data comes over the wire on publications. However, there are some types of data which doesn't need to be persistent or shared between users. For instance, the "logged-in-ness" of the current user, or the route they are currently viewing. 
+
+Although client-side state is often best contained as state of an individual template (and passed down the template heirarchy as arguments where necessary), sometimes you have a need for "global" state that is shared between unrelated sections of the template heirarchy.
+
+Usually such state is stored in a *global singleton* object which we can call a store. A singleton is a data structure of which only a single copy logically exists. The current user and the router from above are typical examples of such global singletons. 
+
+### Types of stores
+
+In Meteor, it's best to make stores *reactive data* sources, as that way they tie most naturally into the rest of the ecosystem. There are a few different packages you can use for stores.
+
+If the store is single-dimensional, you can probably use a `ReactiveVar` to store it (provided by the `reactive-var` package). A `ReactiveVar` has two properties, `get()` and `set()`:
+
+```js
+DocumentHidden = new ReactiveVar(document.hidden);
+$(window).on('visibilitychange', (event) => {
+  DocumentHidden.set(document.hidden);
+});
+```
+
+If the store is multi-dimensional, you may want to use a `ReactiveDict` (from the `reactive-dict` package):
+
+```js
+const $window = $(window);
+const getDimensions = () => {
+  return {
+    width: $window.width(),
+    height: $window.height()
+  };
+};
+
+WindowSize = new ReactiveDict(getDimensions());
+$window.on('resize', () => {
+  WindowSize.set(getDimensions());
+});
+```
+
+The advantage of a `ReactiveDict` is you can access each property individually (`WindowSize.get('width')`), and isolate the reactivity.
```

Comment:
This could be phrased as - it diffs the fields and tracks changes on each field individually.